java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -47,44 +44,44 @@
     }
 
     private final String type;
+    private final String className;
+    private final String pkg;
     private final Map<String, AbstractJSONSchema2Pojo> fields;
     private final Set<String> required;
-    private JObjectOptions options;
+    private final boolean preserveUnknownFields;
 
     public JObject(
+            String pkg,
             String type,
             Map<String, JSONSchemaProps> fields,
             List<String> required,
-            JObjectOptions options) {
-        this.options = options;
+            boolean preserveUnknownFields,
+            String prefix) {
```

Comment:
Could `prefix` be renamed so that is clear it relates to the class? Maybe just add a Javadoc comment here?

Reply:
`classPrefix`